src/components/dialogs/network-modifications/load/modification/load-modification-dialog.js

```diff
@@ -96,6 +112,61 @@ const LoadModificationDialog = ({
         }
     }, [fromEditDataToFormValues, editData]);
 
+    const getConcatenatedProperties = useCallback(
+        (equipment) => {
+            // ex: current Array [ {Object {  name: "p1", value: "v2", previousValue: undefined, added: true, deletionMark: false } }, {...} ]
+            const modificationProperties = getValues(
+                `${ADDITIONAL_PROPERTIES}`
+            );
+            return concatProperties(modificationProperties, equipment);
+        },
+        [getValues]
+    );
+
+    const onEquipmentIdChange = useCallback(
+        (equipmentId) => {
+            if (!equipmentId) {
+                setLoadToModify(null);
+                reset(emptyFormData, { keepDefaultValues: true });
+            } else {
+                setDataFetchStatus(FetchStatus.RUNNING);
+                fetchNetworkElementInfos(
+                    studyUuid,
+                    currentNodeUuid,
+                    EQUIPMENT_TYPES.LOAD,
+                    EQUIPMENT_INFOS_TYPES.FORM.type,
+                    equipmentId,
+                    true
+                )
+                    .then((load) => {
+                        if (load) {
+                            setLoadToModify(load);
+                            reset((formValues) => ({
+                                ...formValues,
+                                [ADDITIONAL_PROPERTIES]:
+                                    getConcatenatedProperties(load),
+                            }));
+                        }
+                        setDataFetchStatus(FetchStatus.SUCCEED);
```

Comment:
So if **!load,** FetchStatus is set to SUCCEED and setLoadToModify is not called ? So the displayed data remains unchanged ?

Reply:
Actually, we do a backendFetch, that does a safeFetch that will throw if the HTTP response is not in 20X, and on this endpoint of study-server it also throws if the load is not found so HTTP response 400-599, sooo if (load) should actually always be true but I couldn't agree more that this conclusion is not straightforward. If we turn stuff to typescript at least we will have a better understanding of stuff like that.